### PR TITLE
Add preference option for unlimited search history

### PR DESCRIPTION
**Pull request by ergor**
- [X] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

This branch adds an option to turn on unlimited search history.

Current behavior: search history is capped at 25 entries when opened, and 3 entries when typing.
New behavior, preference is on: no caps when opening search and no caps when typing.
New behavior, preference is off: as current behavior.
Also, when turning off search history altogether, the new option is disabled.

I have tested and confirmed it works as intended on a virtual device and on my phone.

Screenshot_1569616343

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -90,6 +90,8 @@
     <string name="show_search_suggestions_summary">Show suggestions when searching</string>
     <string name="enable_search_history_title">Search history</string>
     <string name="enable_search_history_summary">Store search queries locally</string>
+    <string name="unlimited_search_history_title">Unlimited search history</string>
+    <string name="unlimited_search_history_summary">Unlimited search history entries</string>
```
Comment (Stypox) [inline app/src/main/res/values/strings.xml]: A better suited description would probably be "Show as many search suggestions as possible from history"

Comment (TobiGr) [reply app/src/main/res/values/strings.xml]: Does this show all suggestions or only the related ones? Maybe "Display all (related) entries from search history" does also fit.

**Comment by TobiGr** [comment]
> But is it ok to add another option to settings? @TobiGr

That's what I was thinking about, too. I do not really see a use-case for this option. How many people want and will use that addition? @theScrabi what's your opinion?

**Comment by ergor** [comment]
My use case is, because I use NewPipe disconnected (not logged in), I get no subscriber feed etc. This way I don't have to involve myself with google. I use my local search history to check up on channels I like, and I rely on it to remember channels as well as music. But, because the local history is limited to the last 25 searches, channels and music I haven't checked out recently just vanishes into the ether.

I wouldn't mind expanding the tweakability of this feature. Also, I wonder: why was the limit of 25 chosen in the first place? Because of performance limitations, or simply preference?

**Comment by gkeegan** [comment]
"Subscribing" to channels on NewPipe doesn't tie you to Google or YouTube accounts at all. It's essentially a browser bookmark.

**Comment by ergor** [comment]
What if instead of this being a new option, just be the way search history works?

**Comment by gkeegan** [comment]
You can make your own playlists in NewPipe. Go to a video, press "Add To", then press " New Playlist" to make a new one. Then you can add to that playlist by doing "Add To" and selecting it. 

Search shouldn't be used as a replacement for playlists, and should be limited.

**Comment by litetex** [comment]
No progress since 2 years, no GitHub actions build and merge conflicts.
Also no feedback to our reviews.

Closing this issue for now. Feel free to reopen it when there is progress again.

**Pull request closed without merging**